### PR TITLE
fix: TimeZone select should show the selected timezone

### LINKED ISSUE #15819: [CAL-4052] Timezone select should show the selected timezone
Timezone select should show the selected value when select box is opened
image

https://www.loom.com/share/142198bce6ea47b7ae881a75c595e925

<sub>[CAL-4052](https://linear.app/calcom/issue/CAL-4052/timezone-select-should-show-the-selected-timezone)</sub>

### PR DESCRIPTION
**Pull request by suprabhu77**
## What does this PR do?

- Fixes #15819
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A-I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [x] N/A-I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist
https://www.loom.com/share/def112516025470d9e29c5818d95cdde?sid=5c95d63b-d809-4a5c-83db-ff059c5c1fb8

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: packages/features/bookings/Booker/components/EventMeta.tsx [hunk 1]
```diff
@@ -165,7 +165,7 @@ export const EventMeta = ({
                     event.lockTimeZoneToggleOnBookingPage ? "cursor-not-allowed" : ""
                   }`}>
                   <TimezoneSelect
-                    menuPosition="fixed"
+                    menuPosition="auto"
```
Comment (anikdhabal) [inline packages/features/bookings/Booker/components/EventMeta.tsx:168]: Nah this is wrong, type `MenuPosition = "fixed" | "absolute"`

Comment (suprabhu77) [reply packages/features/bookings/Booker/components/EventMeta.tsx:168]: @anikdhabal  Yeah, after making it absolute its working fine

Comment (suprabhu77) [reply packages/features/bookings/Booker/components/EventMeta.tsx:168]: Updating the PR

Comment (suprabhu77) [reply packages/features/bookings/Booker/components/EventMeta.tsx]: https://www.loom.com/share/4471b6345ad2488a9e7e27407af89745?sid=ef3c4a96-0b44-4e32-9a95-d547284bcf2f

Comment (suprabhu77) [reply packages/features/bookings/Booker/components/EventMeta.tsx]: @hariombalhara @anikdhabal

**Review by anikdhabal — APPROVED** [review]
LGTM

**Comment by surajgjadhav** [comment]
Hi @anikdhabal , @suprabhu77 ,
We can also improve TimeZoneSelect Component to avoid unnecessary recomputation on each render like this:
image
Please Check
Thanks

**Review by Udit-takkar — APPROVED** [review]
Works fine

**Comment by Udit-takkar** [comment]
> Hi @anikdhabal , @suprabhu77 , We can also improve TimeZoneSelect Component to avoid unnecessary recomputation 

This is a different issue. you can open up a new issue for this

**Pull request merged**